RockWeb/Blocks/Groups/GroupAttendanceDetail.ascx.cs

```diff
@@ -368,9 +326,67 @@ protected void bddlCampus_SelectionChanged( object sender, EventArgs e )
 
         protected void ppAddPerson_SelectPerson( object sender, EventArgs e )
         {
+            string template = GetAttributeValue( "LavaTemplate" );
+
             if ( ppAddPerson.PersonId.HasValue )
             {
-                AddAttendee( ppAddPerson.PersonId.Value );
+                if ( !_attendees.Any( a => a.PersonId == ppAddPerson.PersonId.Value ) )
+                {
+                    var Person = new PersonService( new RockContext() ).Get( ppAddPerson.PersonId.Value );
+                    if ( Person != null )
+                    {
+                        string addPersonAs = GetAttributeValue( "AddPersonAs" );
+                        if ( !addPersonAs.IsNullOrWhiteSpace() && addPersonAs == "Group Member" )
+                        {
+                            var rockContext = new RockContext();
+                            GroupMemberService groupMemberService = new GroupMemberService( rockContext );
+                            GroupTypeRole role = new GroupTypeRoleService( rockContext ).Get( _group.GroupType.DefaultGroupRoleId ?? 0 );
+
+                            var groupMember = new GroupMember { Id = 0 };
+                            groupMember.GroupId = _group.Id;
+
+                            // check to see if the person is already a member of the group/role
+                            var existingGroupMember = groupMemberService.GetByGroupIdAndPersonIdAndGroupRoleId(
+                                _group.Id, Person.Id, _group.GroupType.DefaultGroupRoleId ?? 0 );
+
+                            if ( existingGroupMember != null )
+                            {
+                                return;
```

Comment:
In this situation (leader trying to add a person that's already in the group), their attendance won't be added automatically (checkbox getting checked automatically).  I know this is a low probability scenario, but I just wanted to let you know about it.  I won't change this behavior unless you think it may be an issue.